boards: adafruit: add initial support for esp32s3 feather

leonrinkel opened the pull request:
This PR adds board support for the [Adafruit Feather ESP32S3](https://www.adafruit.com/product/5477) development board. This is basically a copy of the Espressif ESP32-S3-DevKitC board, adapted to accommodate the board-specific features of the Adafruit board, i.e. pinout, feather connector, fuel gauge and on-board NeoPixel. There is an issue #68512 requesting support for this board.

leonrinkel commented:
Rebased on latest main and resized board image to please CI compliance checks.

kartben commented:
@leonrinkel your name sounds familiar 😄

uLipe commented on boards/adafruit/feather_esp32s3/doc/index.rst:
```diff
@@ -0,0 +1,284 @@
+
+Asymmetric Multiprocessing (AMP)
+================================
+
```
Happy to see the AMP being mentioned at board documentation :)

zpm1066 commented:
@leonrinkel Hi,
Is there any chance of also adding support for the Adafruit ESP32S3 TFT Feather (https://www.adafruit.com/product/5483)?

This board is identical to the Feather ESP32S3 but has a built-in ST7789 135x240 display that has TFT/I2C power controlled by a GPIO. 

I’ve been using both boards with esp32s3_devkitc (and esp32s3_devkitm) and external ST7789 but haven’t been able to get the built-in display to work by turning on the TFT power using GPIO hog.

leonrinkel commented:
@zpm1066 I ordered one and will have a look once it arrives 👍

zpm1066 commented:
> @leonrinkel Thank you. It's much appreciated.
I was using compatible = "regulator-fixed" instead of compatible = "power-domain-gpio" to enable power to the TFT Feather board, as in your feather_esp32s3 dts. I'll try again.

zpm1066 commented:
> @leonrinkel 
The morning, I revisited the Adafruit Feather / TFT Feather I2C STEMMA and TFT Feather on-board ST7789 implementation that I did back earlier in Feb. Updated to Zephyr 3.7.x.

Back in Feb, I got I2C STEMMA to work with both the Feather & TFT Feather boards but couldn’t get the on-board TFT ST7789 to work despite the driver init levels being correct for `gpio-hogs_init` , `spi_esp32_init` and `st7789v_init`. 

With Zephyr 3.7, an OLED 128x64 hooked to the ESP32S3 TFT Feather’s I2C STEMMA works fine with my local board `adafruit_esp32s3_tft_feather` using `gpio-hog` on GPIO 21. The same board device tree also with a Feather ESP32S3 if I use SDA = GPIO3 / SCL = GPIO4 & `gpio-hog` on GPIO 7.

Something to note. For the above to work, in “&i2c0” dts overlay, I have `ready-time-ms = <250>;`  /* Min 210 */. The `SSD1306` driver allows to set a delay in init using `read-time-ms` DT property. It is set to 10 ms by default.

I also setup a local device tree for `Adafruit Feather ESP32S3` based on the definitions you setup for the board. 

In order to get the board to work with OLED attached to the I2C STEMMA, I made the following changes.

`adafruit_feather_esp32s3-pinctrl.dtsi`:

	uart0_default: uart0_default {
		group1 {
			pinmux = <UART0_TX_GPIO39>;
			output-high;
		};
		group2 {
			pinmux = <UART0_RX_GPIO38>;
			bias-pull-up;
		};
	};



`adafruit_feather_esp32s3_procpu.dts`:  

The following didn’t work

	/*
	 * On-board regulator powers I2C pull-ups and external devices connected
	 * via the STEMMA QT connector.
	 */
	i2c_reg: i2c_reg {
		compatible = "power-domain-gpio";
		enable-gpios = <&gpio0 7 GPIO_ACTIVE_HIGH>;
	};


However, replacing it with gpio-hog worked.

```
&gpio0 {
	status = "okay";
	i2c-tft-power {
		gpio-hog;
		gpios = <7 GPIO_ACTIVE_HIGH>;
		output-high;
	};
};


```

`adafruit_feather_esp32s3_procpu.dts`:

Setup board to output console & shell-mart to the board’s USB serial port.

```

chosen {
		zephyr,sram = &sram0;
		zephyr,console = &usb_serial;
		zephyr,shell-uart = &usb_serial;
		zephyr,flash = &flash0;
		zephyr,code-partition = &slot0_partition;
		zephyr,bt-hci = &esp32_bt_hci;
	};

...
...

&usb_serial {
	status = "okay";
};


```


Here is the board init sequence:

```

(.venv) quark11:ssd1306_cfb_custom_fonts_v3.7 quark11$ west build -b adafruit_feather_esp32s3/esp32s3/procpu -d build/adafruit_feather_esp32s3_procpu -t initlevels
-- west build: running target initlevels
[0/1] cd /Users/quark11/Documents/2024_zephyr_dev/ssd1306_cfb_cust...ild/adafruit_feather_esp32s3_procpu/zephyr/zephyr.elf --initlevel
EARLY
PRE_KERNEL_1
  __init___device_dts_ord_38: clock_control_esp32_init(__device_dts_ord_38)
  __init_statics_init_pre: statics_init(NULL)
  __init___device_dts_ord_9: gpio_esp32_init(__device_dts_ord_9)
  __init___device_dts_ord_76: gpio_esp32_init(__device_dts_ord_76)
  __init___device_dts_ord_61: uart_esp32_init(__device_dts_ord_61)
  __init___device_dts_ord_64: serial_esp32_usb_init(__device_dts_ord_64)
PRE_KERNEL_2
  __init_sys_clock_driver_init: sys_clock_driver_init(NULL)
POST_KERNEL
  __init_enable_logger: enable_logger(NULL)
  __init_malloc_prepare: malloc_prepare(NULL)
  __init_k_sys_work_q_init: k_sys_work_q_init(NULL)
  __init_gpio_hogs_init: gpio_hogs_init(NULL)
  __init___device_dts_ord_79: i2c_esp32_init(__device_dts_ord_79)
  __init___device_dts_ord_81: ssd1306_init(__device_dts_ord_81)
  __init_enable_shell_uart: enable_shell_uart(NULL)
APPLICATION
SMP


```



The Espressif Monitor output:

```
(.venv) quark11:adafruit_feather_esp32s3_procpu quark11$ west espressif monitor
Serial port /dev/cu.usbmodem14101
Connecting...
Detecting chip type... ESP32-S3
--- idf_monitor on /dev/cu.usbmodem14101 115200 ---
--- Quit: Ctrl+] | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H ---
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x15 (USB_UART_CHIP_RESET),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40048df0
SPIWP:0xee
mode:DIO, clock div:2
load:0x3fc90684,len:0x2148
load:0x40374000,len:0xc664
0x40374000: _WindowOverflow4 at /Users/quark11/zephyrproject370/zephyr/arch/xtensa/core/window_vectors.S:56

SHA-256 comparison failed:
Calculated: c5e49accdc1602848b0a179beae579df8f310563cee58adab033cf043f7ec870
Expected: 0000000010180000000000000000000000000000000000000000000000000000
Attempting to boot anyway...
entry 0x4037aa58
0x4037aa58: __start at /Users/quark11/zephyrproject370/zephyr/soc/espressif/common/loader.c:229

I (77) boot: ESP Simple boot
I (77) boot: compile time Aug 13 2024 10:10:41
W (77) boot: Unicore bootloader
I (77) spi_flash: detected chip: generic
I (79) spi_flash: flash io: dio
W (82) spi_flash: Detected size(4096k) larger than the size in the binary image header(2048k). Using the size in the binary image header.
I (94) boot: chip revision: v0.1
I (97) boot.esp32s3: Boot SPI Speed : 40MHz
I (101) boot.esp32s3: SPI Mode       : DIO
I (104) boot.esp32s3: SPI Flash Size : 4MB
I (108) boot: Enabling RNG early entropy source...
[esp32s3] [INF] DRAM: lma 0x00000020 vma 0x3fc90684 len 0x2148   (8520)
[esp32s3] [INF] IRAM: lma 0x00002170 vma 0x40374000 len 0xc664   (50788)
0x40374000: _WindowOverflow4 at /Users/quark11/zephyrproject370/zephyr/arch/xtensa/core/window_vectors.S:56

[esp32s3] [INF] padd: lma 0x0000e7e8 vma 0x00000000 len 0x1810   (6160)
[esp32s3] [INF] IMAP: lma 0x00010000 vma 0x42000000 len 0x9c2c   (39980)
0x42000000: _stext at ??:?

[esp32s3] [INF] padd: lma 0x00019c34 vma 0x00000000 len 0x63c4   (25540)
[esp32s3] [INF] DMAP: lma 0x00020000 vma 0x3c010000 len 0x351c   (13596)
[esp32s3] [INF] Image with 6 segments
[esp32s3] [INF] DROM segment: paddr=00020000h, vaddr=3c010000h, size=03520h ( 13600) map
[esp32s3] [INF] IROM segment: paddr=00010000h, vaddr=42000000h, size=09C2Ah ( 39978) map

*** Booting Zephyr OS build v3.7.0-734-gc44abdf5b793 ***
[00:00:00.254,000] <inf> main: main_thread
[00:00:00.254,000] <dbg> cfb: cfb_framebuffer_init: number of fonts 2
[00:00:00.348,000] <inf> display: Index[0] font dimensions  5x8
[00:00:00.348,000] <inf> display: Index[1] font dimensions  8x8
[00:00:00.348,000] <inf> display: Selected font: index[3]
[00:00:00.348,000] <inf> display: x_res 128, y_res 64, ppt 8, rows 8, cols 128
[00:00:00.348,000] <inf> display: Zephyr v3.7.99

[00:00:00.348,000] <inf> display: adafruit_feather_esp32s3
uart:~$ device list
devices:
- rtc@60021000 (READY)
- gpio@60004800 (READY)
- gpio@60004000 (READY)
- uart@60000000 (READY)
- uart@60038000 (READY)
- i2c@60013000 (READY)
- ssd1306@3c (READY)
uart:~$ 

```

Next up - Revisit the TFT Feather on-board ST7789 implementation again.

zpm1066 commented:
@leonrinkel 

Update regarding `led_strip` and fuel gauge `MAX17048` on `adafruit_feather_esp32s3`. Also, works with my `adafruit_esp32s3_tft_feather` board.

Both worked fine once I used 

```
&gpio0 {
	status = "okay";
	i2c-tft-power {
		gpio-hog;
		gpios = <7 GPIO_ACTIVE_HIGH>;
		output-high;
	};

	neopixel-power {
		gpio-hog;
		gpios = <21 GPIO_ACTIVE_HIGH>;
		output-high;
	};
};

```

I would like to use `compatible = "power-domain-gpio"` but I couldn't get it work. 

Have you managed to get this compatible type to work with the  `adafruit_feather_esp32s3`? Thank you.

leonrinkel commented:
Ok I managed to get the TFT working and will extend this PR in the coming days. Now that I'm at it, I also got the PSRAM and reverse TFT variants to make sure they'll work too.

zpm1066 commented:
> Ok I managed to get the TFT working and will extend this PR in the coming days. Now that I'm at it, I also got the PSRAM and reverse TFT variants to make sure they'll work too.

Wonderful! Thank you. I'm curious to see what I was doing incorrectly or missing when trying to use the on-board TFT.

btw - I have the 4MB Flash / 2MB PSRAM versions of  Feather ESP32S3, TFT, and a Reverse TFT.

zpm1066 commented:
@leonrinkel Were you able to test out the Adafruit Feather PSRAM and reverse TFT board variants?
Any chance of extending the PR in the next week or so? Thanks.

leonrinkel commented:
- Rebased on latest main
- Corrected some pin numbers that I previously got wrong
- Added the TFT variant and overlay for the `subsys/display/lvgl` sample so it works with this board

I turn on the TFT backlight LED in `board.c` only if the display has been configured, i.e. if the DTS node has status okay. I thought that might be neat, but not sure if this is the proper way.

leonrinkel commented:
Reformatted `board.c` and commit message to please CI checks.

zpm1066 commented:
@leonrinkel Thank you. 

In the Feather TFT variant, pins 39 and 40 are using `gpio1`
```
		dc-gpios = <&gpio1 39 GPIO_ACTIVE_HIGH>;
		reset-gpios = <&gpio1 40 GPIO_ACTIVE_LOW>;
```
		
What prompted the use of `gpio1`, as Adafruit docs for the pinouts have these as `GPIO39` and `GPIO40`?

My Feather TFT didn't work because pins 39 & 40 were using `gpio0`. Once changed these to be on `gpio1`, the display came up fine.

leonrinkel commented:
@zpm1066 The ESP32-S3 has two GPIO controllers `gpio0` for pins 0 through 31 and `gpio1` for the remaining pins 32 to 53.
https://github.com/zephyrproject-rtos/zephyr/blob/7cf124b4a96414dcf72eb564e700e8d3a895f222/dts/xtensa/espressif/esp32s3/esp32s3_common.dtsi#L227-L250

zpm1066 commented:
@leonrinkel Got it! Thanks.
OK, like the Nordic nRF SoCs.

zpm1066 commented:
A question on the lvgl sample, Is there a quick way to invert the display, i.e. white text on a black background? Thanks.

zpm1066 commented:
@leonrinkel For completeness, Do you mind extending the PR to include the Adafruit Reverse Feather TFT varant as well? Thanks.

nordicjm commented on boards/adafruit/feather_esp32s3/doc/index.rst:
```diff
@@ -0,0 +1,284 @@
+Supported Features
+==================
+
+The current Adafruit Feather ESP32S3 board configuration supports the following
```
replace with the board name as used for west build, enclosed with \`\` and call it board target

nordicjm commented on boards/adafruit/feather_esp32s3_tft/doc/img/adafruit_feather_esp32s3_tft.jpg:
convert to webp, then put through https://tinypng.com/

nordicjm commented on boards/adafruit/feather_esp32s3_tft/doc/index.rst:
```diff
@@ -0,0 +1,287 @@
+Supported Features
+==================
+
+The current Adafruit Feather ESP32S3 board configuration supports the following
```
as above

uLipe commented:
@leonrinkel any updates on this?

cujomalainey commented:
If the owner has gone MIA I will happily fork and take over finishing touches to get this landed.

leonrinkel commented:
Sorry will do the requested changes right now

leonrinkel committed:
boards: adafruit: add initial support for esp32s3 feather

The Adafruit Feather ESP32S3 is a development board in the Feather
standard layout, sharing peripheral placement with other devices labeled
as Feathers or FeatherWings. The board is equipped with an ESP32-S3 mini
module, a lithium ion battery charger and a USB-C connector.

Signed-off-by: Leon Rinkel <leon@rinkel.me>

leonrinkel committed:
boards: adafruit: add initial support esp32s3 feather tft

The Adafruit Feather ESP32S3 is a development board in the Feather
standard layout, sharing peripheral placement with other devices labeled
as Feathers or FeatherWings. The board is equipped with an ESP32-S3 mini
module, a lithium ion battery charger and a USB-C connector. Compared to
the base model, this TFT variant additionally comes with a 240x135 pixel
IPS TFT color display.

Signed-off-by: Leon Rinkel <leon@rinkel.me>

raveious commented:
Just playing devil's advocate here, why didn't you implement this as a single board with multiple variants rather than 2 boards?

leonrinkel commented:
From what I read [here](https://docs.zephyrproject.org/latest/glossary.html#term-variant) it seems like the variant qualifier is intended for build variants (e.g. secure/non-secure) rather than board variants. Whereas this is multiple boards, each with their own revision/variant/etc.

nordicjm commented on boards/adafruit/feather_esp32s3/Kconfig.defconfig:
```diff
@@ -0,0 +1,23 @@
+
+if BOARD_ADAFRUIT_FEATHER_ESP32S3_ESP32S3_PROCPU
+
+config HEAP_MEM_POOL_ADD_SIZE_BOARD
+	int
+	default 65535 if WIFI && BT
+	default 51200 if WIFI
+	default 40960 if BT
+	default 4096
```
rename file as `Kconfig`

sylvioalves replied:
Replace all the content in this new file to the below. Current source is old and not needed anymore:
```c
config HEAP_MEM_POOL_ADD_SIZE_BOARD
	int
	default 4096 if BOARD_ADAFRUIT_FEATHER_ESP32S3_ESP32S3_PROCPU
	default 256 if BOARD_ADAFRUIT_FEATHER_ESP32S3_ESP32S3_APPCPU
	```

nordicjm commented on boards/adafruit/feather_esp32s3/doc/index.rst:
```diff
@@ -0,0 +1,283 @@
+Supported Features
+==================
+
+The current ``adafruit_feather_esp32s3`` board target supports the following
```
-target
or add both board targets

nordicjm commented on boards/adafruit/feather_esp32s3/doc/index.rst:
```diff
@@ -0,0 +1,283 @@
+   :goals: build
+
+The usual ``flash`` target will work with the ``adafruit_feather_esp32s3`` board
+configuration. Here is an example for the :zephyr:code-sample:`hello_world`
```
-configuration

nordicjm commented on boards/adafruit/feather_esp32s3_tft/Kconfig.defconfig:
```diff
@@ -0,0 +1,49 @@
+
+if BOARD_ADAFRUIT_FEATHER_ESP32S3_TFT_ESP32S3_PROCPU
+
+config HEAP_MEM_POOL_ADD_SIZE_BOARD
+	int
+	default 65535 if WIFI && BT
+	default 51200 if WIFI
+	default 40960 if BT
+	default 4096
```
move to `Kconfig` and in Kconfig file have this board select `BOARD_LATE_INIT_HOOK`

sylvioalves replied:
Same comment of previous board is applied to this file here.

nordicjm commented on boards/adafruit/feather_esp32s3_tft/Kconfig.defconfig:
```diff
@@ -0,0 +1,49 @@
+
+if BOARD_ADAFRUIT_FEATHER_ESP32S3_TFT_ESP32S3_APPCPU
+
+config HEAP_MEM_POOL_ADD_SIZE_BOARD
+	default 256
+
```
this too

nordicjm commented on boards/adafruit/feather_esp32s3_tft/board.c:
```diff
@@ -0,0 +1,30 @@
+
+static const struct gpio_dt_spec backlight = GPIO_DT_SPEC_GET(DT_ALIAS(backlight), gpios);
+
+static int board_backlight_init(void)
```
`void board_late_init_hook(void)`

nordicjm commented on boards/adafruit/feather_esp32s3_tft/board.c:
```diff
@@ -0,0 +1,30 @@
+
+	return gpio_pin_configure_dt(&backlight, GPIO_OUTPUT_ACTIVE);
+}
+
+SYS_INIT(board_backlight_init, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
+
```
```suggestion
```

nordicjm commented on boards/adafruit/feather_esp32s3_tft/doc/index.rst:
```diff
@@ -0,0 +1,286 @@
+Supported Features
+==================
+
+The current ``adafruit_feather_esp32s3_tft`` board target supports the following
```
as other commit

nordicjm commented on boards/adafruit/feather_esp32s3_tft/doc/index.rst:
```diff
@@ -0,0 +1,286 @@
+   :board: adafruit_feather_esp32s3_tft/esp32s3/procpu
+   :goals: build
+
+The usual ``flash`` target will work with the ``adafruit_feather_esp32s3_tft``
```
as other commit

sylvioalves commented on boards/adafruit/feather_esp32s3/adafruit_feather_esp32s3_appcpu.dts:
```diff
@@ -0,0 +1,31 @@
+ */
+/dts-v1/;
+
+#include <espressif/esp32s3/esp32s3_appcpu.dtsi>
```
Replace this line to:
```c
#include <espressif/esp32s3/esp32s3_wroom_n8.dtsi>
#include <adafruit_qt_py_esp32s3-pinctrl.dtsi>
```

sylvioalves commented on boards/adafruit/feather_esp32s3/adafruit_feather_esp32s3_procpu.yaml:
```diff
@@ -0,0 +1,26 @@
+  - feather_serial
+  - feather_i2c
+  - feather_spi
+testing:
```
You can remove this:
```c
testing:
  ignore_tags:
    - net
    - bluetooth
```

sylvioalves commented on boards/adafruit/feather_esp32s3/doc/index.rst:
```diff
@@ -0,0 +1,283 @@
+LiPo battery charger, a fuel gauge, a USB-C and Qwiic/STEMMA-QT connector. For
+more information, check `Adafruit Feather ESP32S3`_.
+
+.. figure:: img/adafruit_feather_esp32s3.webp
+   :width: 640px
+   :align: center
+   :alt: Board Name
+
+   Adafruit Feather ESP32S3 (Credit: Adafruit Industries)
```
This can also go. Image is automatically added in documentation.

sylvioalves commented on boards/adafruit/feather_esp32s3/doc/index.rst:
```diff
@@ -0,0 +1,283 @@
+  (Tensilica LX7), running at up to 240MHz
+- 512KB SRAM and either 8MB flash or 4MB flash + 2MB PSRAM, depending on the
+  module variant
+- USB-C directly connected to the ESP32 for USB/UART and JTAG debugging
```
`ESP32` -> `ESP32-S3`

sylvioalves commented on boards/adafruit/feather_esp32s3/doc/index.rst:
```diff
@@ -0,0 +1,283 @@
+| GDMA       | on-chip    | dma                                 |
++------------+------------+-------------------------------------+
+| USB-CDC    | on-chip    | serial                              |
++------------+------------+-------------------------------------+
```
Would you mind adding Wi-Fi and BLE entries here?

sylvioalves commented on boards/adafruit/feather_esp32s3_tft/adafruit_feather_esp32s3_tft_appcpu.dts:
```diff
@@ -0,0 +1,31 @@
+ */
+/dts-v1/;
+
+#include <espressif/esp32s3/esp32s3_appcpu.dtsi>
```
#include <espressif/esp32s3/esp32s3_wroom_n8.dtsi>
#include <adafruit_qt_py_esp32s3-pinctrl.dtsi>

sylvioalves commented on boards/adafruit/feather_esp32s3_tft/adafruit_feather_esp32s3_tft_procpu.yaml:
```diff
@@ -0,0 +1,27 @@
+  - feather_i2c
+  - feather_spi
+  - display
+testing:
+  ignore_tags:
+    - net
+    - bluetooth
```
As before, let this go.

sylvioalves commented on boards/adafruit/feather_esp32s3_tft/doc/index.rst:
```diff
@@ -0,0 +1,286 @@
+.. zephyr:board:: adafruit_feather_esp32s3_tft
+
+Overview
```
Please, apply the same comments from previous board in this one.

sylvioalves commented on boards/adafruit/feather_esp32s3/adafruit_feather_esp32s3_procpu.dts:
```diff
@@ -0,0 +1,206 @@
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
```
Make it `sram1` instead: `zephyr,sram = &sram1;`

sylvioalves commented on boards/adafruit/feather_esp32s3/adafruit_feather_esp32s3_appcpu.dts:
```diff
@@ -0,0 +1,32 @@
+	compatible = "adafruit,feather_esp32s3", "espressif,esp32s3";
+
+	chosen {
+		zephyr,sram = &sram0;
```
Make it sram1 instead: zephyr,sram = &sram1;

sylvioalves commented on boards/adafruit/feather_esp32s3_tft/Kconfig:
```diff
@@ -0,0 +1,39 @@
+	default 4096 if BOARD_ADAFRUIT_FEATHER_ESP32S3_TFT_ESP32S3_PROCPU
+	default 256 if BOARD_ADAFRUIT_FEATHER_ESP32S3_TFT_ESP32S3_APPCPU
+
+if BOARD_ADAFRUIT_FEATHER_ESP32S3_TFT
```
Please move all the below to `Kconfig.defconfig` file instead.

sylvioalves replied:
`if BOARD_ADAFRUIT_FEATHER_ESP32S3_TFT_ESP32S3_PROCPU`

sylvioalves commented on boards/adafruit/feather_esp32s3_tft/adafruit_feather_esp32s3_tft_appcpu.dts:
```diff
@@ -0,0 +1,32 @@
+	compatible = "adafruit,feather_esp32s3_tft", "espressif,esp32s3";
+
+	chosen {
+		zephyr,sram = &sram0;
```
Same as before: `sram1`

sylvioalves commented on boards/adafruit/feather_esp32s3_tft/adafruit_feather_esp32s3_tft_procpu.dts:
```diff
@@ -0,0 +1,253 @@
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
```
Same as before: `sram1`

simonguinot commented on samples/drivers/led/led_strip/boards/adafruit_feather_esp32s3_procpu.conf:
```diff
@@ -0,0 +1,4 @@
+CONFIG_POWER_DOMAIN=y
+CONFIG_GPIO=y
+CONFIG_PM_DEVICE=y
+CONFIG_PM_DEVICE_POWER_DOMAIN=y
```
Is all this necessary to demonstrate the `led-strip` driver on this board ? For example, I can see that `CONFIG_GPIO` is already enabled in the board defconfig and as far I can tell it is not a dependency of the WS2812 LED strip driver...
